util/db/db.go

```diff
@@ -93,11 +95,34 @@ type db struct {
 
 // NewDB returns a new instance of the argo database
 func NewDB(namespace string, settingsMgr *settings.SettingsManager, kubeclientset kubernetes.Interface) ArgoDB {
-	return &db{
+	dbInstance := db{
 		settingsMgr:   settingsMgr,
 		ns:            namespace,
 		kubeclientset: kubeclientset,
 	}
+	clusterSecrets, err := dbInstance.listSecretsByType(common.LabelValueSecretTypeCluster)
+	if err != nil {
+		log.WithError(err).Errorln("could not list secrets by type")
+		return &dbInstance
+	}
+	dbSettings, err := dbInstance.settingsMgr.GetSettings()
+	if err != nil {
+		log.WithError(err).Errorln("could not get DB settings")
+		return &dbInstance
+	}
+	for _, clusterSecret := range clusterSecrets {
+		cluster, err := secretToCluster(clusterSecret)
+		if err != nil {
+			log.Errorf("could not unmarshal cluster secret %s", clusterSecret.Name)
+			continue
+		}
+		if cluster.Server == appv1.KubernetesInternalAPIServerAddr {
+			if !dbSettings.InClusterEnabled {
+				log.Warnf("cluster %q uses in-cluster server address but it's disabled in Argo CD settings", cluster.Name)
+			}
+		}
+	}
```

Comment:
Looks like the singular purpose of this block is to log a warning. Maybe pull into a function which does solely that? No need for a return type, it can be called just for the side-effect.

Reply:
Done. Thanks!